File: src/main/java/com/jcloisterzone/figure/RangeFigure.java
package com.jcloisterzone.figure;

import com.jcloisterzone.game.state.GameState;
import com.jcloisterzone.game.state.PlacedTile;
import io.vavr.collection.Stream;

public interface RangeFigure {

    Stream<PlacedTile> getRangeTiles(GameState state);

}

File: src/main/ts/com/jcloisterzone/figure/RangeFigure.ts
import type { Stream } from "../../../io/vavr/SeqTypes.js";
import type { GameState } from "../game/state/GameState.js";
import type { PlacedTile } from "../game/state/PlacedTile.js";

/** A figure that affects a range of tiles (dragon, fairy, ...). */
export interface RangeFigure {
  getRangeTiles(state: GameState): Stream<PlacedTile>;
}
